services/thermal_mgr/thermal_mgr.c

```diff
@@ -20,52 +20,80 @@ static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
 
 static QueueHandle_t thermalMgrQueueHandle;
 static StaticQueue_t thermalMgrQueueBuffer;
-static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
+static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH *
+                                          THERMAL_MGR_QUEUE_ITEM_SIZE];
 
 static void thermalMgr(void *pvParameters);
 
 void initThermalSystemManager(lm75bd_config_t *config) {
-  memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
-  memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
-  thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+    memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
+    memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
 
-  memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
-  memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
+    thermalMgrTaskHandle = xTaskCreateStatic(
+        thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE, config, 1,
+        thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
-  thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
+    memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
+    memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
+    thermalMgrQueueHandle = xQueueCreateStatic(
+        THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+        thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+    if (event == NULL) return ERR_CODE_INVALID_ARG;
+
+    if (xQueueSend(thermalMgrQueueHandle, (void *)event, (TickType_t)0) ==
+        errQUEUE_FULL) {
+        return ERR_CODE_QUEUE_FULL;
+    }
 
-  return ERR_CODE_SUCCESS;
+    return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
-  /* Implement this function */
+    thermal_mgr_event_t interruptEvent;
+    interruptEvent.type = THERMAL_MGR_EVENT_INTERRUPT_CMD;
+    (void)thermalMgrSendEvent(&interruptEvent);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
-  while (1) {
-    
-  }
+    if (pvParameters == NULL) return;
+
+    lm75bd_config_t config = *(lm75bd_config_t *)pvParameters;
+    thermal_mgr_event_t taskEvent;
+    error_code_t taskStatus;
+    float temp = 0;
```

Comment:
You can limit the scope of these variables since it looks like they'd only be needed inside the `if (xQueueReceive(thermalMgrQueueHandle, &taskEvent, portMAX_DELAY) == pdTRUE) {` block.